### PR TITLE
feat: localDatastore support for unsorted distance queries

### LINKED ISSUE #1571: withinKilometers withinMiles withinRadians not working with LocalDatastore
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description

Unsorted distance queries aren't supported on the LocalDatastore. Unsorted distance queries uses the `$centerSphere` + `$geoWithin` operators found [here](https://github.com/parse-community/Parse-SDK-JS/blob/alpha/src/ParseQuery.js#L1644)

### Steps to reproduce

```
it(`LocalDatastore support unsorted distance queries`, async () => {
      const object = new TestObject();
      const firstPoint = new Parse.GeoPoint({ latitude: 40.0, longitude: -30.0 });
      object.set({ location: firstPoint });
      await object.save();
      await object.pin();

      const sorted = false;
      const query = new Parse.Query(TestObject);
      query.withinKilometers(
        'location',
        new Parse.GeoPoint({ latitude: 40.0, longitude: -30.0 }),
        2,
        sorted
      );
      query.fromLocalDatastore();
      const results = await query.find();
      assert.equal(results.length, 1); // Returns 0 objects
});
```
### Actual Outcome

No objects are returned

### Expected Outcome

Should return objects within a certain radius and distance

###  Environment

Server
- Parse Server version: `5.3.0-alpha.27`
- Operating system: `Mac OS 12.6`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Localhost`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `4.4.0`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `Localhost`

Client
- Parse JS SDK version: `3.4.4`

### PR DESCRIPTION
**Pull request by dplewis**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).

### Issue Description

Related issue: https://github.com/parse-community/Parse-SDK-JS/issues/1571

### TODOs before merging

- [x] Add tests

### CODE REVIEW

**Comment by mtrezza** [comment]
@dplewis If click on "Details" for the failing "Commit Message" test, you'll see that we don't currently use a scope in the commit message. The scope needs to be incorporated into the commit summary. Also, a PR needs to reference an issue in the same repo. That issue can then reference an issue in another repo.

File: src/OfflineQuery.js [hunk 1]
```diff
@@ -513,9 +513,23 @@ function matchesKeyConstraints(className, object, objects, key, constraints) {
...
+      if (compareTo.$polygon) {
+        const points = compareTo.$polygon.map(geoPoint => [
+          geoPoint.latitude,
+          geoPoint.longitude,
+        ]);
+        const polygon = new ParsePolygon(points);
+        return polygon.containsPoint(object[key]);
+      }
+      // $centerSphere
+      const [WGS84Point, maxDistance] = compareTo.$centerSphere;
```
Comment (mtrezza) [inline src/OfflineQuery.js]: Would it make sense to also wrap this into an if statement? Just like `if (compareTo.$polygon) {`

Comment (dplewis) [reply src/OfflineQuery.js]: Done

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.0-alpha.5](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.0-alpha.5)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [3.5.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/3.5.0)

**Pull request merged**